File: examples/stream/subscribe/node/subscribeToATopic.js
const { StreamService } = require("m3o/stream");

// Subscribe to messages for a given topic.
async function subscribeToAtopic() {
  let streamService = new StreamService(process.env.MICRO_API_TOKEN);
  let rsp = await streamService.subscribe({
    topic: "events",
  });
  console.log(rsp);
}

subscribeToAtopic();

File: examples/stream/subscribe/go/subscribeToATopic.go
package example

import (
	"fmt"
	"github.com/micro/services/clients/go/stream"
	"os"
)

// Subscribe to messages for a given topic.
func SubscribeToAtopic() {
	streamService := stream.NewStreamService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := streamService.Subscribe(&stream.SubscribeRequest{
		Topic: "events",
	})
	fmt.Println(rsp, err)
}
